test/Engine/ProtoTest/Associative/MethodResolution.cs

```diff
@@ -388,6 +388,30 @@ public void MAGN5729_Repro_Simple()
             thisTest.Verify("r3", new object[] { true, false });
         }
 
+        [Test]

```

Comment:
@aparajit-pratap good question. I'll test if this case is fixed as well. But as `foo = 2` is a definition, we should create a variable `foo` which shadows function name `foo`.